core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/pushpull/PullJobDelegate.java

```diff
@@ -272,6 +272,14 @@ protected String doExecuteProvisioning(
             }
         }
 
+        if (pullTask.getResource().getProvisionSorter() != null) {
```

Comment:
Two problems here:

1. when `pullTask.getResource().getProvisionSorter() == null` the `DefaultProvisionSorter` shall be used
1. (most important) we need to browse the sorted provisions, not to sort the provisions inplace